### PR TITLE
DA kzg custom blob sizes

### PR DESCRIPTION
**Pull request by danielSanchezQ**
* Extract and refactor library methods to use custom elements per blob
* Added benches

### CODE REVIEW

File: nomos-da-core/kzg/src/dynamic_kzg.rs [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+};
+use kzg_traits::{Fr, Poly, G1};
+
+pub fn blob_to_kzg_commitment(
+    blob: &Blob,
+    s: &FsKZGSettings,
+    field_elements_per_blob: usize,
+) -> FsG1 {
+    let mut out = FsG1::default();
+    g1_linear_combination(&mut out, &s.secret_g1, &blob.inner, field_elements_per_blob);
```
[the remark this answers is not in the document]
Comment (danielSanchezQ) [reply nomos-da-core/kzg/src/dynamic_kzg.rs:24]: Yes, eip implementation uses some constants. The idea is to substitute them to make it flesible if possible. We are hitting some walls anyway and we will need help with this. For now we will aim for double eip field elements (8192) which we know works.

File: nomos-da-core/kzg/src/types.rs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        self.0.to_bytes()
+    }
+}
+
+impl Blob {
+    pub fn from_bytes(data: &[u8], settings: &KzgSettings) -> Result<Self, Box<dyn Error>> {
+        let mut inner = Vec::with_capacity(data.len() / settings.bytes_per_field_element);
+        for chunk in data.chunks(settings.bytes_per_field_element) {
+            inner.push(FsFr::from_bytes(chunk)?);
+        }
```
Comment (youngjoon-lee) [inline nomos-da-core/kzg/src/types.rs:42]: What if the last chunk is shorter than `bytes_per_field_element`? Then, it seems `FsFr::from_bytes(chunk)` will fails.

Comment (danielSanchezQ) [reply nomos-da-core/kzg/src/types.rs:42]: Oh yes, actually I had a padding in an earlier version.

**Review by youngjoon-lee — APPROVED** [review]
👍 I didn't fully understand logics in `nomos-da-core/kzg/src/lib.rs`, but it seems they were copied from `blst/src/eip_4844.rs`.

**Pull request merged**